xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererDX.cpp

```diff
@@ -219,14 +219,15 @@ COverlayImageDX::COverlayImageDX(CDVDOverlayImage* o, CRect& rSource)
     m_x = (0.5f * o->width + o->x) / o->source_width;
     m_y = (0.5f * o->height + o->y) / o->source_height;
 
-    int videoSourceH{static_cast<int>(rSource.Height())};
-    int videoSourceW{static_cast<int>(rSource.Width())};
+    float subRatio{static_cast<float>(o->source_width) / o->source_height};
+    float vidRatio{rSource.Width() / rSource.Height()};
 
-    if ((o->source_height == videoSourceH || videoSourceH % o->source_height == 0) &&
-        (o->source_width == videoSourceW || videoSourceW % o->source_width == 0))
+    // We always consider aligning 4/3 subtitles to the video,
+    // for example SD DVB subtitles (4/3) can be rendered on a fullhd video
+    bool isSubResSquare{subRatio == 1.25f || subRatio == 1.33f};
+
+    if (subRatio == vidRatio || isSubResSquare)
```

Comment:
This may not even work in all cases: float numbers should not be treated as if they were exact values.

A range should be compared to avoid rounding errors:
`a == b` -----> `abs(a - b) < 0.001`

Reply:
from my test not result rounding errors, the ratio values result always correct
but no problem for me add suggested change

Reply:
`abs` but return always 0
https://rextester.com/STGYN71172
whats wrong?

Reply:
abs works with integers afaik, so if you pass floats they‘re cut. is it `fabs()` ?

Reply:
thanks now seem to works!

Reply:
change applied